shenyu-admin/src/main/java/org/apache/shenyu/admin/service/impl/TagServiceImpl.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shenyu.admin.service.impl;
+
+import com.google.common.collect.Lists;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.shenyu.admin.mapper.TagMapper;
+import org.apache.shenyu.admin.model.dto.TagDTO;
+import org.apache.shenyu.admin.model.entity.TagDO;
+import org.apache.shenyu.admin.model.query.TagQuery;
+import org.apache.shenyu.admin.model.vo.TagVO;
+import org.apache.shenyu.admin.service.TagService;
+import org.apache.shenyu.admin.utils.Assert;
+import org.springframework.stereotype.Service;
+
+/**
+ * Implementation of the {@link org.apache.shenyu.admin.service.TagService}.
+ */
+@Service
+public class TagServiceImpl implements TagService {
+
+    private TagMapper tagMapper;
+
+    public TagServiceImpl(final TagMapper tagMapper) {
+        this.tagMapper = tagMapper;
+    }
+
+    @Override
+    public int create(final TagDTO tagDTO) {
+        TagDO tagRelationDO = TagDO.buildTagDO(tagDTO);
```

Comment:
It is suggested that we construct ext ourselves, and store the full json data of the parent tag.